instagram-app/src/components/CommentSection/CommentSection.js

```diff
@@ -1,116 +1,134 @@
 import React from "react";
+import { CommentStyling } from "../Styling";
+import { CommentStyling2 } from "../Styling";
+import { CommentStyling3 } from "../Styling";
+import { CommentBoxStyling } from "../Styling";
+import styled, { css } from "styled-components";
 
 class CommentSection extends React.Component {
   //boom! constructor gone!
   state = {
-comments: this.props.comments,
-newComment: ""
-  }
+    comments: this.props.comments,
+    newComment: ""
+  };
+
+  /* 
 
-/* 
+addNewComment function called on submit.
+
+nutshell: "value" field of input box - what user types into - is set to send all
+the user typing to this.state above, as this.state.comment.  addNewComment function
+just grabs whatever's in this.state.newComment, shoves it into text in the template
+ and then this.setState below that shoves the template into the comment's array
+ at the end (newCommentAdded), the second newComment just resets the text field to zero
 
-addNewComment called on submit.  Says,  "take a look in value field of input - whatever it is, stick it into this.state.newComment above"
-text on const = this.state.newComment, so that gets pulled in when addnewCommentruns
-then setState takes all existing this.state.comments, puts it into a new array, and adds this.state.newComment as a new array item at the end
-the second newComment: "" just resets the value field of hte input box back to zero for the next comment
-and that's how the addnewcomment/submit button works
 */
 
-addNewComment = e => {
-  e.preventDefault();
-  //don't refresh page^
-  //now declare new comment format
-  //NOT the same newComment as above or name, can be different!
-  const newComment = {
-    username: "Cannonball Run",
-    text: this.state.newComment
-    //text here is pointing to the newCOmment variable from state
-  }
-  //the abvoe just set the template, now to update and push it in
+  addNewComment = e => {
+    e.preventDefault();
+    //says "don't refresh page^"
+    //now declare new comment format
+    //NOT the same newComment as above or name, can be different!
+    const newCommentAddedByUser = {
+      username: "Cannonball Run",
+      text: this.state.newComment
```

Comment:
This username needs to be dynamic and determined by whoever is logged in. You would need a location in you state to hold and use that data when adding a new comment.